go/cmd/vpnkit-userland-proxy/proxy.go

```diff
@@ -43,17 +43,25 @@ var vSockUDPPortOffset = 0x20000
 
 // From docker/libnetwork/portmapper/proxy.go:
 
+type localBind string
+
+const (
+	bestEffortLocalBind = localBind("best-effort")
+	alwaysLocalBind     = localBind("true")
+	neverLocalBind      = localBind("false")
+)
+
 // parseHostContainerAddrs parses the flags passed on reexec to create the TCP or UDP
 // net.Addrs to map the host and container ports
-func parseHostContainerAddrs() (host net.Addr, port int, container net.Addr, localIP bool) {
+func parseHostContainerAddrs() (host net.Addr, port int, container net.Addr, bind localBind) {
 	var (
 		proto         = flag.String("proto", "tcp", "proxy protocol")
 		hostIP        = flag.String("host-ip", "", "host ip")
 		hostPort      = flag.Int("host-port", -1, "host port")
 		containerIP   = flag.String("container-ip", "", "container ip")
 		containerPort = flag.Int("container-port", -1, "container port")
 		interactive   = flag.Bool("i", false, "print success/failure to stdout/stderr")
-		noLocalIP     = flag.Bool("no-local-ip", true, "bind only on the Host, not in the VM")
+		local         = flag.String("no-local-ip", string(bestEffortLocalBind), "bind only on the Host, not in the VM")
```

Comment:
The flag `--no-local-ip` doesn't really make sense now I think.

Reply:
That's true. There are a couple of scripts floating around that have `-no-local-ip` (without an argument) which I'd like to keep working for a little while. I've pushed an update with
- `-no-local-ip` removed from the help text
- a new `-local-bind (best effort | always | never)` argument
- a special-case search of the arguments for `-no-local-ip` as a flag which sets the default to `never` as a temporary workaround for the existing scripts

I propose merging something like this, then separately modifying the scripts and when those have been released, coming back here and removing the workaround. WDYT?

Reply:
SGTM.